PyRDF/backend/Dist.py

```diff
@@ -669,7 +667,9 @@ def reducer(mergeables_out, mergeables_in):
                 for filename in value:
                     snapshot_chain.Add(filename)
                 # Create a new rdf with the chain and return that to user
-                snapshot_rdf = PyRDF.RDataFrame(snapshot_chain)
+                factory = PyRDF.DataFrame.DataFrameFactory()
```

Comment:
I guess this might change depending on the changes suggested at the beginning, regarding the relocation of backend-dependent code.